### ISSUE TITLE
xgraph Subgraphx files has typo resulting in erroneous test accuracy

### ISSUE THREAD

**Issue by ahoefn** [divelab/DIG#255]
### Description
In following [file](https://github.com/divelab/DIG/blob/main/dig/xgraph/SubgraphX/models/train_gnns.py) in the xgraph/Subgraphx directory of the main branch, there seems to be a typo on line 82. Here, the test accuracy is calculated using the code
`    test_state, _, _ = test_GC(dataloader['train'], gnnNets, criterion)`
The test_GC function then calculates accuracy and the loss. However, the 'test' subset of the data should be used here instead (see e.g. line 170 of the same file).

### Suggested fix
Change 'train' to 'test'.
